src/sentry/static/sentry/app/views/eventsV2/table/index.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import {Location} from 'history';
+import {browserHistory} from 'react-router';
+import styled from 'react-emotion';
+
+import {Client} from 'app/api';
+import {Organization} from 'app/types';
+import withApi from 'app/utils/withApi';
+
+import Pagination from 'app/components/pagination';
+
+import {DEFAULT_EVENT_VIEW_V1} from '../data';
+import EventView from '../eventView';
+import TableView from './tableView';
+import {TableData} from './types';
+
+type TableProps = {
+  api: Client;
+  location: Location;
+  organization: Organization;
+};
+type TableState = {
+  isLoading: boolean;
+  error: null | string;
+
+  eventView: EventView;
+  pageLinks: null | string;
+
+  tableData: TableData | null | undefined;
+};
+
+/**
+ * `Table` is a container element that handles 2 things
+ * 1. Fetch data from source
+ * 2. Handle pagination of data
+ *
+ * It will pass the data it fetched to `TableView`, where the state of the
+ * Table is maintained and controlled
+ */
+class Table extends React.PureComponent<TableProps, TableState> {
+  static getDerivedStateFromProps(props: TableProps, state: TableState): TableState {
+    return {
+      ...state,
+      eventView: EventView.fromLocation(props.location),
+    };
+  }
+
+  state: TableState = {
+    isLoading: true,
+    error: null,
+
+    eventView: EventView.fromLocation(this.props.location),
+    pageLinks: null,
+
+    tableData: null,
+  };
+
+  componentDidMount() {
+    const {location} = this.props;
+
+    if (!this.state.eventView.isValid()) {
+      const nextEventView = EventView.fromEventViewv1(DEFAULT_EVENT_VIEW_V1);
+
+      browserHistory.replace({
+        pathname: location.pathname,
+        query: {
+          ...location.query,
+          ...nextEventView.generateQueryStringObject(),
+        },
+      });
+      return;
+    }
+
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.location !== prevProps.location) {
```

Comment:
Could we be more specific here? Perhaps comparing the fields, aliases, query text and globalselection parameters?

Reply:
This is from the previous implementation of `Table`. I've checked that the reference of the `Location` object changes when it's updated so this will work, but I'll add in a few more checks to make it more robust.